parser.go

```diff
@@ -94,9 +97,11 @@ type TemplateFieldSpecifier struct {
 type Session struct {
 	buffers *sync.Pool
 
-	mut       sync.RWMutex
-	templates [][]TemplateFieldSpecifier
-	minRecord []uint16
+	mut                  sync.RWMutex
+	templates            [][]TemplateFieldSpecifier
+	minRecord            []uint16
+	templateIdMap        map[uint16]uint16
```

Comment:
Capitalize ID in Go, i.e., `templateIDs` (typically the fact that it's a map is not spelled out)`. Also in other places that refer to IDs.